reconcile/typed_queries/saas_files.py

```diff
@@ -75,6 +76,13 @@ class Config:
         # ignore `namespaceSelector` and 'provider' fields from the GQL schema
         extra = Extra.ignore
 
+    def uid(self, saas_file_name: str, resource_template_name: str) -> str:
```

Comment:
Does it make sense to have these args as attributes instead? I.e., when creating `SaasResourceTemplateTarget` also add the parents `SaasResourceTemplate.name` and `SaasFile.name`?

Reply:
Unfortunately, this isn't easy to achieve. See https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/typed_queries/saas_files.py#L248 how the `SaasFile` gets composed.

Reply:
Oh I see.

In that case, does it make sense to extend the `export_model` function? It is only used for the purpose of converting gql model data to our rendered saas file data right?

```python
class SaasResourceTemplateTarget(ConfiguredBaseModel):
    ...
    # By not specifying the field parameter, we should be fine if those are missing in dict data
    parent_saas_name: Optional[str]
    parent_template_name: Optional[str]


def export_model(model: SaasFileV2) -> SaasFile:
    saas_file = SaasFile(**convert_parameters_to_json_string(model.dict(by_alias=True)))
    for tmpl in saas_file.templates:
        for target in tmpl.targets:
            target.parent_saas_name = saas_file.name
            target.parent_template_name = tmpl.name
    return saas_file
```

Not 100% sure it is worth the effort - might as well be Im completely missing sth here. My naive feeling here is that it would be nice for the consumer to not care about saas_file name etc anylonger and just be able to call `.uid()`.

Reply:
Please take a look at https://github.com/app-sre/qontract-reconcile/pull/3473/commits/cd8bc893ee72f8ec3b90c57e08a37fd2577ea4d5